src/seedu/addressbook/data/person/Address.java

```diff
@@ -12,8 +12,9 @@
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    public Object[] values;
```

Comment:
If you are intending to store the objects (block. street ...), you don't need an array. You can just keep the object references within the class itself.